### PR TITLE
Fixes #50

### LINKED ISSUE #50: Create new endpoint to calculate parameters
Create a single endpoint that can be run when the "Calculate Missing Parameters" button is clicked in the StreamStats storm runoff model.

Inputs:
- lat
- lon
- curveNumberMethod: "runoff" or "area"
- TcMethod: "traveltime" or "lagtime"
- length (will be the LFPLENGTH BC)-- Null by default (since Tc Travel Time Method doesn't need this)
- slope (will be the CSL10_85fmBC)-- Null by default (since Tc Travel Time Method doesn't need this)
- dataSheetFlow-- Null by default (since Tc Lag Time Equation doesn't need this)
- dataExcessSheetFlow-- Null by default (since Tc Lag Time Equation doesn't need this)
- dataShallowConcentratedFlow-- Null by default (since Tc Lag Time Equation doesn't need this)
- dataChannelizedFlowOpenChannel-- Null by default (since Tc Lag Time Equation doesn't need this)
- dataChannelizedFlowStormSewer-- Null by default (since Tc Lag Time Equation doesn't need this)
- dataChannelizedFlowStormSewerOrOpenChannelUserInputVelocity-- Null by default (since Tc Lag Time Equation doesn't need this)

Outputs:
- Rainfall Distribution Curve letter
- Tc
- PRF
- CN
- S
- Ia

The endpoint will use the following functions:
- `rainfallDistributionCurve`
- `travelTimeMethodTimeOfConcentration` or `lagTimeMethodTimeOfConcentration` (depending on TcMethod)
- `PRFData`
- `weightedCurveNumber`

Will help with https://github.com/USGS-WiM/StreamStats/issues/1385

### PR DESCRIPTION
**Pull request by harper-wavra**
Created calculateMissingParameters endpoint which combines rainfallDistributionCurve, PRFData, weightedCurveNumber, and travelTimeMethodTimeOfConcentration or lagTimeMethodTimeOfConcentration (depending on TcMethod) into single function.

Included is an insomia json to help with testing. 

[insomiaCollection.zip](https://github.com/USGS-WiM/SC-RunoffModelingServices/files/9202556/insomiaCollection.zip)

### CODE REVIEW

File: main.py [hunk 1]
```diff
@@ -554,4 +686,35 @@ def scsyntheticunithydrograph(request_body: SCSyntheticUnitHydrograph, response:
         }
 
     except Exception as e:
-        raise HTTPException(status_code = 500, detail =  str(e))
\ No newline at end of file
+        raise HTTPException(status_code = 500, detail =  str(e))
+
+@app.post("/calculatemissingparameters/")
+def calculatemissingparameters(request_body: CalculateMissingParameters, response: Response):
```
Comment (amedenblik) [inline main.py:692]: Maybe rename the endpoint to clarify that it's from SC Synthetic UH method specifically? `calculateMissingParametersSCSUH` or something?

File: main.py [hunk 2]
```diff
@@ -554,4 +686,35 @@ def scsyntheticunithydrograph(request_body: SCSyntheticUnitHydrograph, response:
...
+            request_body.dataSheetFlow,
+            request_body.dataExcessSheetFlow,
+            request_body.dataShallowConcentratedFlow,
+            request_body.dataChannelizedFlowOpenChannel,
+            request_body.dataChannelizedFlowStormSewer,
+            request_body.dataChannelizedFlowStormSewerOrOpenChannelUserInputVelocity,
+        )
+        return {
+            "rainfall_distribution_curve_letter": rainfall_distribution_curve_letter,
+            "time_of_concentration": Tc,
```
Comment (amedenblik) [inline main.py:712]: I got the following response body:

```
{
  "rainfall_distribution_curve_letter": "B",
  "time_of_concentration": [
    64.46157606015156,
    null
  ],
  "peak_rate_factor": 240,
  "curve_number": 66.9098326632795,
  "S": 4.945486488248321,
  "Ia": 0.9890972976496641
}
```

Do you know why Tc is returning an array with `null`? I think it should just be a single number. Apologies if this is due to an error I caused.

**Review by amedenblik — CHANGES_REQUESTED** [review]
Looks good! Made 2 small comments.

File: main.py (hunk 2 above)
Comment (harper-wavra) [reply main.py:712]: Its the warning message being returned: https://github.com/USGS-WiM/SC-RunoffModelingServices/blob/ad5203ba08a39397e0daa0df7d857c7c2486c3c8/Tc_Calculator.py#L255. I wasn't sure if I should grab that or not.

Comment (harper-wavra) [reply main.py:712]: I could create a separate return variable for the warning message, that might be easier when it comes to working with it in the client?

Comment (amedenblik) [reply main.py:712]: Oh sorry, I was looking at the wrong function return 🤦‍♀️. Let's keep it as-is.

Comment (harper-wavra) [reply main.py:712]: Added warningMessage.

**Review by amedenblik — APPROVED** [review]

**Pull request merged**